sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -1353,9 +1353,15 @@ void simd_obj_impl<T, N, T1, SFINAE>::copy_to(
       if constexpr (RemN == 1) {
         Addr[NumChunks * ChunkSize] = Tmp[NumChunks * ChunkSize];
       } else if constexpr (RemN == 8 || RemN == 16) {
-        simd<uint32_t, RemN> Offsets(0u, sizeof(T));
-        scatter<UT, RemN>(Addr + (NumChunks * ChunkSize), Offsets,
-                          Tmp.template select<RemN, 1>(NumChunks * ChunkSize));
+        if constexpr (sizeof(T) == 1) {
+          simd<int32_t, N / 4> BC = Tmp.template bit_cast_view<int32_t>();
```

Comment:
Shouldn't this use 'RemN / 4' instead of 'N / 4' ?

Reply:
Also, can you say, why this work-around code is needed? Is it because GPU RT doesn't handle scatter<char, 8> properly?

Reply:
This code inside copy_to() calls copy_to(). It is not the recursion because the data type is different, but it would be good if this code was a bit more obvious by using the code chosen for that second copy_to() call, e.g. block_store if it would be lowered to it.

Reply:
> of scatter<char, 8> on GPU
IMO, if it is GPU RT error, then it is better to report it to GPU RT team than adding this additional code/work-around.

Also, it is unclear if the fix is complete, for example it looks like memory.hpp(Line 1374) may call scatter<char,8> (for RemN==7) or scatter<char,16> (for RemN==15) and cause the same error, right?

Reply:
@vmutya, can you please take a look at this case. Do you think it is GPU RT error, or SYCL calls scatter incorrectly?

Reply:
If it is a _temporary_ fix here, then it needs a TODO comment near the code for the particular case handling asking to remove it when GPU RT issue is fixed.

Reply:
Please create an internal tracker, but do not add it to the comment.